src/move_mode.cpp

```diff
@@ -75,6 +75,8 @@ void move_mode::load( const JsonObject &jo, std::string_view/*src*/ )
     optional( jo, was_loaded, "move_speed_multiplier", _move_speed_mult, 1.0 );
     optional( jo, was_loaded, "mech_power_use", _mech_power_use, 2 );
     optional( jo, was_loaded, "swim_speed_mod", _swim_speed_mod, 0 );
+    //Make sure no json modders make 0 or negative bwm, this avoids divbyzeros
+    optional( jo, was_loaded, "bash_weight_modifier", _bash_weight_modifier, 1.0 );
```

Comment:
Sorry, I wasn't clear. You can just enforce this with a reader.

```suggestion
    optional( jo, was_loaded, "bash_weight_modifier", _bash_weight_modifier, numeric_bounds_reader( 0.0 ), 1.0 );
```

Reply:
> Sorry, I wasn't clear. You can just enforce this with a reader.

numeric_bounds_reader is undefined, and i have no idea how to work with it :(

Reply:
`numeric_bound_reader` (extra s, my bad) from generic_factory.h.

Reply:
nice ty